### PR TITLE
refactor: events meta app support custom common events and components

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

### What is the new behavior?

将事件绑定弹窗body中左右两部分提取出单独的组件，方便替换成自定义组件
复杂的数据通讯使用了provide和inject

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced various new Vue components for managing and selecting event binding methods, enhancing the event configuration interface.
  - Added functionalities for dynamic method generation and validation within the event binding dialogs.

- **Refactor**
  - Consolidated and optimized import statements across multiple components for better code readability and maintainability.
  - Updated component logic and UI structures for improved user experience in event configuration.

- **Bug Fixes**
  - Corrected import paths and updated method assignment strategies to ensure proper functionality.

- **Style**
  - Made cosmetic changes in comments and formatting across various files to maintain code cleanliness and readability.

### CODE REVIEW

File: packages/settings/events/src/components/BindEventsMethods.vue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+export default {
+  components: {
+    TinySearch: Search,
+    IconYes: iconYes()
+  },
+  inheritAttrs: false,
+  props: {
+    eventBinding: {
+      type: Object,
+      default: () => {}
```
Comment (chilingling) [inline packages/settings/events/src/components/BindEventsMethods.vue]: `() => ({})`，这样默认值才是 empty object

Comment (gene9831) [reply packages/settings/events/src/components/BindEventsMethods.vue]: done

**Review by chilingling — DISMISSED** [review]

File: packages/settings/events/src/components/BindEventsDialogBodyLeft.vue [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      }
+
+      return name
+    }
+
+    const selectMethod = (data) => {
+      context.bindMethodInfo = data
+    }
+
+    watchEffect(() => {})
```
Comment (chilingling) [inline packages/settings/events/src/components/BindEventsDialogBodyLeft.vue]: 这个是不是没用了？

Comment (gene9831) [reply packages/settings/events/src/components/BindEventsDialogBodyLeft.vue]: 已删除

**Review by chilingling — DISMISSED** [review]

File: packages/settings/events/src/components/BindEventsDialog.vue [hunk 3]
```diff
@@ -134,79 +50,22 @@ export default {
     }
   },
   setup(props) {
+    const { BindEventsDialogSidebar, BindEventsDialogContent } = getMergeMeta('engine.setting.event').components
```
Comment (hexqi) [inline packages/settings/events/src/components/BindEventsDialog.vue]: 引用自己的id可以import meta.js拿到

Comment (gene9831) [reply packages/settings/events/src/components/BindEventsDialog.vue]: updated

**Review by hexqi — APPROVED** [review]

**Review by hexqi — APPROVED** [review]

**Review by hexqi — DISMISSED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**